File: Runtime/TransitionSpreadDistribution.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2009-2015 ApexRMS.
'
'************************************************************************************

''' <summary>
''' Transition Spread Distribution
''' </summary>
''' <remarks></remarks>
Friend Class TransitionSpreadDistribution

    Private m_TransitionSpreadDistributionId As Integer
    Private m_StratumId As Nullable(Of Integer)
    Private m_Iteration As Nullable(Of Integer)
    Private m_Timestep As Nullable(Of Integer)
    Private m_TransitionGroupId As Integer
    Private m_StateClassId As Integer
    Private m_MinimumDistance As Double
    Private m_MaximumDistance As Double
    Private m_RelativeAmount As Double
    Private m_Proportion As Double

    Public Sub New(
        ByVal transitionSpreadDistributionId As Integer,
        ByVal stratumId As Nullable(Of Integer),
        ByVal iteration As Nullable(Of Integer),
        ByVal timestep As Nullable(Of Integer),
        ByVal transitionGroupId As Integer,
        ByVal stateClassId As Integer,
        ByVal maximumDistance As Double,
        ByVal relativeAmount As Double)

        Me.m_TransitionSpreadDistributionId = transitionSpreadDistributionId
        Me.m_StratumId = stratumId
        Me.m_Iteration = iteration
        Me.m_Timestep = timestep
        Me.m_TransitionGroupId = transitionGroupId
        Me.m_StateClassId = stateClassId
        Me.m_MaximumDistance = maximumDistance
        Me.m_RelativeAmount = relativeAmount

    End Sub

    ''' <summary>
    ''' Gets the Transition Spread Distribution Id
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public ReadOnly Property TransitionSpreadDistributionId As Integer
        Get
            Return Me.m_TransitionSpreadDistributionId
        End Get
    End Property

    ''' <summary>
    ''' Gets the stratum Id
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public ReadOnly Property StratumId As Nullable(Of Integer)
        Get
            Return Me.m_StratumId
        End Get
    End Property

    ''' <summary>
    ''' Gets the iteration
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public ReadOnly Property Iteration As Nullable(Of Integer)
        Get
            Return Me.m_Iteration
        End Get
    End Property

    ''' <summary>
    ''' Gets the timestep
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public ReadOnly Property Timestep As Nullable(Of Integer)
        Get
            Return Me.m_Timestep
        End Get
    End Property

    ''' <summary>
    ''' Gets the transition group Id
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public ReadOnly Property TransitionGroupId As Integer
        Get
            Return Me.m_TransitionGroupId
        End Get
    End Property

    ''' <summary>
    ''' Gets the state class Id
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public ReadOnly Property StateClassId As Integer
        Get
            Return Me.m_StateClassId
        End Get
    End Property

    ''' <summary>
    ''' Gets or sets the minimum distance
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Friend Property MinimumDistance As Double
        Get
            Return Me.m_MinimumDistance
        End Get
        Set(value As Double)
            Me.m_MinimumDistance = value
        End Set
    End Property

    ''' <summary>
    ''' Gets the maximum distance
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public ReadOnly Property MaximumDistance As Double
        Get
            Return Me.m_MaximumDistance
        End Get
    End Property

    ''' <summary>
    ''' Gets the relative amount
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public ReadOnly Property RelativeAmount As Double
        Get
            Return Me.m_RelativeAmount
        End Get
    End Property

    ''' <summary>
    ''' Gets or sets the proportion
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property Proportion As Double
        Get
            Return Me.m_Proportion
        End Get
        Set(value As Double)
            Me.m_Proportion = value
        End Set
    End Property

End Class

File: src/Runtime/TransitionSpreadDistribution.cs
// stsim: A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2019 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

namespace SyncroSim.STSim
{
    internal class TransitionSpreadDistribution
    {
        private int m_TransitionSpreadDistributionId;
        private int? m_StratumId;
        private int? m_Iteration;
        private int? m_Timestep;
        private int m_TransitionGroupId;
        private int m_StateClassId;
        private double m_MinimumDistance;
        private double m_MaximumDistance;
        private double m_RelativeAmount;
        private double m_Proportion;

        public TransitionSpreadDistribution(
            int transitionSpreadDistributionId, int? stratumId, int? iteration, int? timestep, 
            int transitionGroupId, int stateClassId, double maximumDistance, double relativeAmount)
        {
            this.m_TransitionSpreadDistributionId = transitionSpreadDistributionId;
            this.m_StratumId = stratumId;
            this.m_Iteration = iteration;
            this.m_Timestep = timestep;
            this.m_TransitionGroupId = transitionGroupId;
            this.m_StateClassId = stateClassId;
            this.m_MaximumDistance = maximumDistance;
            this.m_RelativeAmount = relativeAmount;
        }

        public int TransitionSpreadDistributionId
        {
            get
            {
                return this.m_TransitionSpreadDistributionId;
            }
        }

        public int? StratumId
        {
            get
            {
                return this.m_StratumId;
            }
        }

        public int? Iteration
        {
            get
            {
                return this.m_Iteration;
            }
        }

        public int? Timestep
        {
            get
            {
                return this.m_Timestep;
            }
        }

        public int TransitionGroupId
        {
            get
            {
                return this.m_TransitionGroupId;
            }
        }

        public int StateClassId
        {
            get
            {
                return this.m_StateClassId;
            }
        }

        internal double MinimumDistance
        {
            get
            {
                return this.m_MinimumDistance;
            }
            set
            {
                this.m_MinimumDistance = value;
            }
        }

        public double MaximumDistance
        {
            get
            {
                return this.m_MaximumDistance;
            }
        }

        public double RelativeAmount
        {
            get
            {
                return this.m_RelativeAmount;
            }
        }

        public double Proportion
        {
            get
            {
                return this.m_Proportion;
            }
            set
            {
                this.m_Proportion = value;
            }
        }
    }
}
